com.unity.shadergraph/Editor/Data/Nodes/UV/TriplanarNode.cs

```diff
@@ -122,13 +168,15 @@ public virtual void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode gene
                         , GetVariableNameForNode()
                         , GetSlotValue(NormalInputId, generationMode));
 
-                    sb.AppendLine("$precision4 {0} = $precision4(normalize({1}_X.zyx * {1}_Blend.x + {1}_Y.xzy * {1}_Blend.y + {1}_Z.xyz * {1}_Blend.z), 1);"
-                        , GetVariableNameForSlot(OutputSlotId)
-                        , GetVariableNameForNode());
-                    sb.AppendLine("$precision3x3 {0}_Transform = $precision3x3(IN.WorldSpaceTangent, IN.WorldSpaceBiTangent, IN.WorldSpaceNormal);", GetVariableNameForNode());
-                    sb.AppendLine("{0}.rgb = TransformWorldToTangent({0}.rgb, {1}_Transform);"
-                        , GetVariableNameForSlot(OutputSlotId)
+                    var outputVariable = GetVariableNameForSlot(OutputSlotId);
+                    sb.AppendLine("$precision4 {0} = $precision4({1}_X.zyx * {1}_Blend.x + {1}_Y.xzy * {1}_Blend.y + {1}_Z.xyz * {1}_Blend.z, 1);"
+                        , outputVariable
                         , GetVariableNameForNode());
+
+                    // transform the normal from input to output space, and normalize
+
+                    outputVariable = $"{outputVariable}.rgb";
+                    SpaceTransformUtil.GenerateTransformCodeStatement(normalTransform, outputVariable, outputVariable, sb);
```

Comment:
Should this have `outputVariable` for both the input and output?

Reply:
or:  outputVariable.rgb = transform(outputVariable.rgb);   in this case